torch/csrc/jit/pickler.cpp

```diff
@@ -547,12 +618,32 @@ void Unpickler::readList() {
     for (auto it = stack_.begin() + start; it != stack_.end(); ++it) {
       list->elements().emplace_back(it->ivalue().toInt());
     }
-  } else {
+  } else if (list_ivalue.ivalue().isTensorList()) {
+    auto list = stack_.at(start - 1).ivalue().toTensorList();
```

Comment:
Might be nice to template this to reduce copy-pasta

Reply:
I'd rather save the templating for a follow up more general clean up of the Pickler class and API

Reply:
Eh i think 4 lines copy pasta'd in 3 different places should be fixed in this PR

Reply:
Since bool needs a specialization, templating this part ends up with more code than is already here and adds the extra layer of indirection

Reply:
why does bool need a specialization here ? i believe bool list only needs a specialization if you're trying to pass a reference to an element in a vector<bool>, which you're not doing here

Reply:
I am ok with this organization, as long as the weird but repeated parts are pulled to the top:
for instalce:

```
auto list_ivalue = stack_.at(start -1).ivalue(); 
auto elements = ArrayRef(stack_).slice(start);
```